allennlp/commands/train.py

```diff
@@ -129,7 +133,7 @@ def train_model(param_dict: Dict[str, Any]):
     train_data = dataset_reader.read(train_data_path)
 
     # TODO(Mark): work out how this is going to be built with different options.
-    vocab = Vocabulary.from_dataset(train_data)
+    vocab = Vocabulary.from_dataset(train_data, non_padded_namespaces=non_padded_namespaces)
```

Comment:
Yeah, this is what Mark meant in his TODO comment.  We need to add a `from_params` option here, and add a `vocab` key to the top-level config.  It's a little complicated, though, because of different options for instantiating this object (e.g., we need something like `from_dataset_and_params`, but done in a sane way that doesn't have too many ways of creating the object), so I wouldn't worry about fixing it in this PR.  I'll add a github issue for this.